x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowNodeTask.java

```diff
@@ -61,13 +61,14 @@
     private final BiConsumer<TimeValue, Runnable> scheduler;
 
     private volatile long lastRequestedSeqno;
-    private volatile long leaderGlobalCheckpoint;
+    // package-protected visibility for testing only:
+    volatile long leaderGlobalCheckpoint;
```

Comment:
I am not sure if we should increase the visibility to allow changing this volatile variable?

Reply:
To be honest, I'm not happy with it too. But otherwise testing the isolated steps is much trickier.

Reply:
@dnhatn What do you think about this: 47bc37c4d1de1e0915c13580f9fe074de0c2e2a9 ?

Reply:
isn't this exposed in the status object? can't we use that maybe? just an idea

Reply:
@bleskes The test sets the leaderGlobalCheckpoint. I just changed how this test now sets it via this: 47bc37c4d1de1e0915c13580f9fe074de0c2e2a9